ast.hh

```diff
@@ -52,8 +52,10 @@ template <class T> void debug_log(const char *msg, int depth, T *obj)
 			depth, msg,
 		demangled ? demangled : mangled, obj);
 	free(static_cast<void*>(demangled ? demangled : buffer));
-#endif // DEBUG_AST_CONSTRUCTION
 }
+#else
+template <class T> void debug_log(const char *, int /* depth */, T *) {}
+#endif // DEBUG_AST_CONSTRUCTION
```

Comment:
If we put the `#ifdef` back inside the function body, we still end up with a `-Wunused-parameter` warning. We could fix that with an "unused by release" macro...